### PR TITLE
feat: LSDV-E-131: Labels inside regions

### PR DESCRIPTION
**Pull request by Travis1282**
PR fulfills these requirements
	•	  Commit message(s) and PR title follows the format [fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made ex. fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors
	•	  Tests for the changes have been added/updated (for bug fixes/features)
	•	  Docs have been added/updated (for bug fixes/features)
	•	  Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
	•	  Self-reviewed and ran all changes on a local instance (for bug fixes/features)
Change has impacts in these area(s)
(check all that apply)
	•	  Product design
	•	  Frontend
Describe the reason for change
There were not labels on the regions
What does this fix?
There are not optionally labels on the regions for singular and multiple labels
What is the new behavior?
labels can be optionally switched on and off in the editor ui
What is the current behavior?
no labels
What libraries were added/updated?
N/A
Does this change affect performance?
N/A
Does this change affect security?
N/A
What alternative approaches were there?
N/A
What feature flags were used to cover this change?
N/A
Does this PR introduce a breaking change?
(check only one)
	•	  Yes, and covered entirely by feature flag(s)
	•	  Yes, and covered partially by feature flag(s)
	•	  No
	•	  Not sure (briefly explain the situation below)
What level of testing was included in the change?
(check all that apply)
	•	  e2e
	•	  integration
	•	  unit

### CODE REVIEW

File: src/tags/object/AudioUltra/model.js [hunk 1]
```diff
@@ -142,6 +142,7 @@ export const AudioModel = types.compose(
             const labels = activeState?.selectedValues();
 
             selectedRegions.forEach(r => {
+              console.log('selectedRegions', r, selectedColor, labels);
```
Comment (bmartel) [inline src/tags/object/AudioUltra/model.js:145]:
```suggestion
```

File: src/tags/object/AudioUltra/model.js [hunk 2]
```diff
@@ -339,6 +340,8 @@ export const AudioModel = types.compose(
         // area id is assigned to WS region during deserealization
           const find_r = self.annotation.areas.get(wsRegion.id);
 
+          console.log('addRegion', wsRegion, find_r);
```
Comment (bmartel) [inline src/tags/object/AudioUltra/model.js:343]:
```suggestion
```

File: src/lib/AudioUltra/Regions/Region.ts [hunk 3]
```diff
@@ -32,6 +32,57 @@ export class Region extends Segment {
...
+    if (this.labels?.length && this.controller.showLabels) {
+      const layer = this.controller.layerGroup;
+      const color = this.color;
+      const timelineTop = this.timelinePlacement;
+      const timelineLayer = this.visualizer.getLayer('timeline');
+      const timelineHeight = this.timelineHeight;
+      const top = timelineLayer?.isVisible && timelineTop ? timelineHeight : 0;
+      const labelMeasures = this.labels.map((label) => layer.context.measureText(label));
+
+      const height = labelMeasures.reduce((accumulator, currentValue) => {
```
Comment (bmartel) [inline src/lib/AudioUltra/Regions/Region.ts:53]: How does this work with multiple labels? I forgot to check that case, but is this stacking the labels or taking the average height of all the text heights? Maybe add a comment about the approach used here.

**Review by bmartel — CHANGES_REQUESTED** [review]

File: src/lib/AudioUltra/Regions/Region.ts (hunk 3 above)
Comment (Travis1282) [reply src/lib/AudioUltra/Regions/Region.ts:53]: It stacks them, so the reduce calculates the combined height of the box and the heightCalc in the labels for each determines the vertical position of the text

**Review by bmartel — APPROVED** [review]

**Pull request merged**